### DISCUSSION TITLE
Issue with Modbus rules when using Suricata

### DISCUSSION THREAD

**Discussion by Katulich** [Security-Onion-Solutions/securityonion#3942] [Unsupported Versions] [answered]
Hello Everyone!
My use case with IDS Security Onion is monitoring and logging communication within industrial control system, more precise SCADA system.
I am using Security Onion V2.3.30, standalone version and have VM Analyst installed.  Security Onion is installed as virtual machine using VMWare pro v 16.1.1 as Cent OS 7.
When I write custom SURICATA rules ( e.g. picture_1) none of the rules pop up from Suricata within SO. 
![picture_1](https://user-images.githubusercontent.com/82998588/115671977-5d697880-a34b-11eb-8d69-59662e75a27f.png)

I add rules within Analyst VM - /opt/so/saltstack/local/salt/idstools and then edit local.rules.
after adding rules, I write the following:

_sudo salt-call state.highstate
sudo so-rule-update
sudo salt-call state.apply suricata_

Other rules (e.g. _tcp any any -> any 502 (...)_ that are added in local.rules work, but are not triggered every time.
I have enabled modbus by editing suricata.yaml using SALT:
/opt/so/saltstack/local/pillar/ and then editing global.sls by adding text from defaults.yaml (https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/suricata/defaults.yaml) and editing modbus enable parameter to "yes" within added text. After editing global.sls I run this line: sudo salt-call state.highstate
When checking suricata.yaml file in /opt/so/conf/suricata my modbus module is enabled (picture_2)
![picture_2](https://user-images.githubusercontent.com/82998588/115671918-4d519900-a34b-11eb-903f-215228d350dd.png)

Am I missing some steps?
With regards,
Katulich

**Comment by Katulich** [comment, marked as answer]
I have found a solution to this problem.
The problem was that if I started my Security Onion before i started PLC-s, there were no logs.
If the PLC-s were started after the Security Onion, everything was ok and i was able to get Suricata alerts.
With regards,
Katulich

**Reply by kornflake003-GITHUB** [reply]
Hi @Katulich, I need help with the syntax in the global.sls. ¿Could you show me the part of the global.sls file where you told it to activate the Modbus protocol?.
Thanks.

**Comment by Katulich** [comment]
Hello kornflake003,
You need to configure Suricata
https://docs.securityonion.net/en/2.3/suricata.html#configuration
in a way that in the:
/opt/so/saltstack/local/pillar/global.sls or
in /opt/so/saltstack/local/pillar/minions/$SENSORNAME_$ROLE.sls  you add
the following text:
suricata:
  config:
    app-layer:
      protocols:
        modbus:
          #request-flood: 500
          enabled: "yes"
          detection-ports:
            dp: 502
          stream-depth: 0

You need to watch on the syntax.
After that, you need to save the global.sls and then in the manager run:
sudo salt-call state.highstate
After that, your modbus module should run.
Bye,
Katulich

sri, 15. pro 2021. u 02:55 kornflake003-GITHUB ***@***.***>
napisao je:

**Reply by kornflake003-GITHUB** [reply]
Thanks a lot for your support. From now on, i hope to see some alerts from modbus protocol.
@Katulich
